src/api/API.cpp

```diff
@@ -0,0 +1,32 @@
+
+/*
+ * Copyright 2007-2024, RTE (https://www.rte-france.com)
+ * See AUTHORS.txt
+ * SPDX-License-Identifier: MPL-2.0
+ * This file is part of Antares-Simulator,
+ * Adequacy and Performance assessment for interconnected energy networks.
+ *
+ * Antares_Simulator is free software: you can redistribute it and/or modify
+ * it under the terms of the Mozilla Public Licence 2.0 as published by
+ * the Mozilla Foundation, either version 2 of the License, or
+ * (at your option) any later version.
+ *
+ * Antares_Simulator is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * Mozilla Public Licence 2.0 for more details.
+ *
+ * You should have received a copy of the Mozilla Public Licence 2.0
+ * along with Antares_Simulator. If not, see <https://opensource.org/license/mpl-2-0/>.
+ */
+
+#include "API.h"
+
+namespace Antares::API
+{
+SimulationResults APIInternal::run(IStudyLoader* study_loader) {
+    study_ = study_loader->load();
```

Comment:
As it is now, we see that the loader has a much broader responsibility than just loading a study: it's also responsible for running a simulation.

We either need to rename it, or to restrict what it actually does.
From a high level view, one possibility seems to be to:
1. push down the study loader to `Application` function that runs the simulation
2. use the `Application` here in the `run` method

Another, probably better possibility would be to invert:
1. Move all "business logic" code from `Application` to another layer `Simulation` for example (similar to your APIInternal, but meant to be used not only by the public API)
2. Use the loader interface in that layer
3. Call this simulation layer from application, with the file tree study loader
4. Call also this simulation layer from API internal

Reply:
Out dated.
New code for information and more inline with suggestion:
`SimulationResults APIInternal::run(IStudyLoader* study_loader)
{
    study_ = study_loader->load();
    return execute();
}`